### ISSUE TITLE
Red Baron / Pokey finding + need someone w/ Pokey on a breadboard to help.

### ISSUE THREAD

**Issue by dinkc64** [mamedev/mame#4067]
This was reposted from PM talks between me and Haze, he suggested that I post it here in the case that someone can help with the pokey+breadboard research.
--- begin message ---
I found something nice.  Red Baron (in bzone) has been broken since the .80's in MAME.  What caused it to break is this addition to pokey.c[pp]:
[code]
	case ALLPOT_C:
		/****************************************************************
		 * If the 2 least significant bits of SKCTL are 0, the ALLPOTs
		 * are disabled (SKRESET). Thanks to MikeJ for pointing this out.
		 ****************************************************************/
		if( (m_SKCTL & SK_RESET) == 0) <----- This block.
		{
			data = 0;
			LOG(("POKEY '%s' ALLPOT internal $%02x (reset)\n", tag(), data));
		}
[/code]

The part where it returns 0 in the SKRESET state is what broke the game.  

so, the game keeps the chip in SKRESET mode until a new game is started.  that means for ALLPOT: 0 is returned during attract, and when game is started SKRESET is unactive and it goes up to 0x80 (analogue center-point).  This causes the game's internal calibration to go completely out-of-whack and cause the plane to go in a neverending right circle.

Now there's 2 ways we can fix it:
first, remove the return 0 on SKRESET until we can get someone with a pokey on a breadbord to actual verify this.  
I tried to do some research into MikeJ's pokey work and can't find anything that leads to this conclusion.  Perhaps it wasn't talked about publically.

Since the first way is controversial until proven, this way is even worse and more hacky: we can read the ALLPOT right from the memmap and bypass the pokey altogether.

So basically before we do anything, we need someone to put a pokey on a breadboard, hook up some pot's, continuously read the ALLPOT register in the SKRESET state and normal state and report the findings.  Do you know someone who could help with this?

best regards,
- dink

**Comment by galibert** [comment]
It could just be that the reset value is incorrect.  What happens if the
return is changed to 0x80?

  OG.

**Comment by dinkc64** [comment]
If the reset value is changed to 0x80, the game will be fine.  But, it seems very unlikely that the ALLPOT register would return 0x80 in SKRESET state.  I think it's more likely that it returns 0xff in SKRESET than 0 or 0x80.  But 0xff causes breakage, too..  (sometimes it won't be apparent until the second game w/0xff)

**Comment by MooglyGuy** [comment]
Has this now been fixed? If so, please close the issue.
